src/main/java/semantics/DirectStatementDeleter.java

```diff
@@ -0,0 +1,293 @@
+
+import static semantics.RDFImport.RELATIONSHIP;
+
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader.InvalidCacheLoadException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.Callable;
+import org.eclipse.rdf4j.model.BNode;
+import org.eclipse.rdf4j.model.Statement;
+import org.eclipse.rdf4j.rio.RDFHandlerException;
+import org.neo4j.graphdb.Direction;
+import org.neo4j.graphdb.GraphDatabaseService;
+import org.neo4j.graphdb.Label;
+import org.neo4j.graphdb.Node;
+import org.neo4j.graphdb.Relationship;
+import org.neo4j.graphdb.RelationshipType;
+import org.neo4j.helpers.collection.Iterables;
+import org.neo4j.logging.Log;
+
+/**
+ * Created by jbarrasa on 09/11/2016.
+ */
+
+class DirectStatementDeleter extends RDFToLPGStatementProcessor implements Callable<Integer> {
+
+  public static final Label RESOURCE = Label.label("Resource");
+  public static final String[] EMPTY_ARRAY = new String[0];
+  Cache<String, Node> nodeCache;
+  private String bNodeInfo;
+  private int notDeletedStatementCount;
+
+  public DirectStatementDeleter(GraphDatabaseService db, long batchSize, long nodeCacheSize,
+      int handleUrls, int handleMultivals, Set<String> multivalPropUriList,
+      boolean keepCustomDataTypes,
+      Set<String> customDataTypedPropList, Set<String> predicateExclusionList,
+      boolean typesToLabels, boolean klt,
+      String languageFilter, boolean applyNeo4jNaming, Log l) {
+
+    super(db, languageFilter, handleUrls, handleMultivals, multivalPropUriList, keepCustomDataTypes,
+        customDataTypedPropList, predicateExclusionList, klt,
+        typesToLabels, applyNeo4jNaming, batchSize);
+    nodeCache = CacheBuilder.newBuilder()
+        .maximumSize(nodeCacheSize)
+        .build();
+    log = l;
+    bNodeInfo = "";
+    notDeletedStatementCount = 0;
+  }
+
+  @Override
+  public void endRDF() throws RDFHandlerException {
+    Util.inTx(graphdb, this);
+    totalTriplesMapped += mappedTripleCounter;
+    if (handleUris == 0) {
+      addNamespaceNode();
+    }
+
+    log.info("Successful (last) partial commit of " + mappedTripleCounter + " triples. " +
+        "Total number of triples deleted is " + totalTriplesMapped + " out of "
+        + totalTriplesParsed + " parsed.");
+  }
+
+  private void addNamespaceNode() {
+    Map<String, Object> params = new HashMap<>();
+    params.put("props", namespaces);
+    graphdb.execute("MERGE (n:NamespacePrefixDefinition) SET n+={props}", params);
+  }
+
+  public Map<String, String> getNamespaces() {
+    return namespaces;
+  }
+
+  // Stolen from APOC :)
+  private Object toPropertyValue(Object value) {
+    if (value instanceof Iterable) {
+      Iterable it = (Iterable) value;
+      Object first = Iterables.firstOrNull(it);
+      if (first == null) {
+        return EMPTY_ARRAY;
```

Comment:
Since the returned array is modifiable, posterior calls might be compromised.
Please return `new String[0]` and remove the static variable